### ISSUE TITLE
Should x-properties allow multiple values?

### ISSUE THREAD

**Issue by ksuther** [libical/libical#592]
Apple uses unescaped commas in their x-property values. For example:

```
X-APPLE-TRAVEL-START;VALUE=URI:geo:99.000000,-99.000000
X-CALENDARSERVER-ATTENDEE-COMMENT;X-CALENDARSERVER-ATTENDEE-REF="urn:x-uid
 :XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX";X-CALENDARSERVER-DTSTAMP=20220101T
 000000Z:Test, test.
```

Reading RFC 5545, my interpretation is that we shouldn't be splitting these into multiple values during parsing.

https://www.rfc-editor.org/rfc/rfc5545#section-3.1.2:
> ... However, it should be noted that some properties
   support encoding multiple values in a single property by separating
   the values with a COMMA character.  Individual property definitions
   should be consulted for determining whether a specific property
   allows multiple values and in which of these two forms. ...

The way I read it, only properties that explicitly say they can contain multiple values should be split.

If that's the case, x-properties should be ignored in `parser_get_next_value()` and `icalmemory_strdup_and_quote()`. Any opinions (@winterz @ksmurchison)?

**Comment by ksmurchison** [comment]
I think your reading is correct.  That said, the current multiple value handling is broken, and there are outstanding issues to fix it #75 #77 #79 .
I think the [vCard](https://github.com/libical/libical/pull/584) handling library that I wrote does a good job with multi-valued properties and parameters.  I will try to port that to iCalendar when I get time.

**Comment by ksuther** [comment]
Sounds good. For the short term I'll make a PR to exclude x-properties from multiple values and escaping and add some tests.

**Comment by winterz** [comment]
fixed with 8b0117e01a04ad57cdb0652819eea181e2a3645f
